### ISSUE TITLE
What usages of %in%+class() should be linted for expect_s3_class_linter()?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2482]
Here's an earlier test that was deactivated:

```r
expect_lint(
  "expect_true('lm' %in% class(x))",
  "expect_s3_class\\(x, k\\) is better than expect_equal\\(class\\(x\\), k",
  expect_s3_class_linter
)
```

That test looks good... what about `k %in% class(x)` or `class(x) %in% k` in general? Does `k` have to be literal?
